Gems/ROS2/Code/Source/RobotImporter/SDFormat/ROS2SDFormatHooksUtils.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ROS2SDFormatHooksUtils.h"
+#include <ROS2/Communication/TopicConfiguration.h>
+#include <VehicleDynamics/WheelControllerComponent.h>
+
+#include <sdf/Joint.hh>
+
+namespace ROS2::SDFormat
+{
+    void HooksUtils::AddTopicConfiguration(
+        SensorConfiguration& sensorConfig, const AZStd::string& topic, const AZStd::string& messageType, const AZStd::string& configName)
+    {
+        TopicConfiguration config;
+        config.m_topic = topic;
+        config.m_type = messageType;
+        sensorConfig.m_publishersConfigurations.insert(AZStd::make_pair(configName, config));
+    }
+
+    AZ::EntityId HooksUtils::GetJointEntityId(
+        const std::string& jointName, const sdf::Model& sdfModel, const CreatedEntitiesMap& createdEntities)
+    {
+        const auto jointPtr = sdfModel.JointByName(jointName);
+        if (jointPtr != nullptr)
+        {
+            const auto linkName(jointPtr->ChildName().c_str());
+            const auto linkPtr = sdfModel.LinkByName(linkName);
+            if (linkPtr != nullptr && createdEntities.contains(linkPtr))
+            {
+                const auto& entityResult = createdEntities.at(linkPtr);
+                return entityResult.IsSuccess() ? entityResult.GetValue() : AZ::EntityId();
+            }
+        }
+
+        return AZ::EntityId();
+    }
+
+    void HooksUtils::EnableMotor(const AZ::EntityId& entityId)
+    {
+        AZ::Entity* entity = nullptr;
+        AZ::ComponentApplicationBus::BroadcastResult(entity, &AZ::ComponentApplicationRequests::FindEntity, entityId);
+        if (entity != nullptr)
+        {
+            PhysX::EditorHingeJointComponent* jointComponent = entity->FindComponent<PhysX::EditorHingeJointComponent>();
+            if (jointComponent != nullptr)
+            {
+                jointComponent->Activate();
```

Comment:
I'm not too fond of activation to change something in the component state.
PhysX::EditorHingeJointComponent is dependent on service : `PhysicsDynamicRigidBodyService`
Please consider :
- Checking after activation if the component is **really** active and performing early return and error message.
- Call the Handler interface directly without activation, looking at the ` EditorHingeJointComponent::SetBoolValue` implementation, it is nothing more than a setter.
Btw is component EditorHingeJointComponent active at this state ?